packages/engine/paima-funnel/src/funnels/emulated/funnel.ts

```diff
@@ -249,6 +251,8 @@ export class EmulatedBlocksFunnel extends BaseFunnel {
       return newEntry.processingQueue;
     })();
 
+    const [chainName, _config] = await GlobalConfig.mainEvmConfig();
```

Comment:
Well, there is always a main evm funnel at the base of the stack of funnels. And the actual "blocks" that are received by this funnel are essentially the ones for the main chain. That said, I'm actually not sure how this is supposed to work, especially during the presync, since I haven't really used the emulated funnel that much.

In this case I'm not really using the config, I'm just using the chain name, and this is only used in order to know which of the events that come from the earlier funnels needs to be used.

That said, maybe this is incorrect? Maybe there should be an emulated funnel PER evm funnel instead? And each one should process only the blocks from that particular network. Is that what you are suggesting?